### PR TITLE
Deploy to s3

### PR DESCRIPTION
**Pull request by elleneub**
Adds scripts to build, deploy to s3, and invalidate cloudfront cache. I also updated the team photos so I could test if the redeploy was working. I made a run where it did the whole build and deploy here: https://github.com/emergentworks/website/runs/4302542051?check_suite_focus=true

Afterwards I switched it back to only do that on merge of pull request to main. Sorry to put so many jira tickets in one PR!

Photos ticket: https://emergentworks.atlassian.net/browse/WEB-13
Local deploy ticket: https://emergentworks.atlassian.net/browse/WEB-23
Github actions ticket: https://emergentworks.atlassian.net/browse/WEB-26

### CODE REVIEW

File: .github/workflows/github-actions-demo.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: GitHub Actions Demo
```
Comment (markthom-as) [inline .github/workflows/github-actions-demo.yml:1]: I think we should change the YAML file name and action name from the defaults if possible

File: README.md [hunk 2]
```diff
@@ -13,3 +13,24 @@ setup RSA key for your github account (example found here: https://jhooq.com/git
...
+
+### Prerequistes
+
+To do this you must:
+
+- Have s3 and cloudfront permission for the emergentworks.org s3 bucket and the E3NHW48S3W8DOR cloudfront distribution.
+- Install the aws cli https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
+  - If you have a homebrew you can alternatley use 'brew install awscli'
+- Get your AWS public key and secret access key, and set them up as explained here: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
+  - To do this you may need to request those keys from an AWS admin
```
Comment (markthom-as) [inline README.md:27]: can we add "by emailing devops@emergentworks.org" please!

File: deploy-website.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# Build the new website into the 'public' folder
+gatsby build
```
Comment (markthom-as) [inline deploy-website.sh:4]: What happens here if the build fails? Does the script exit, or does it still delete the S3 files?
I think we want to make sure the script exits here on failure

File: deploy-website.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+# Delete old contents of the s3 buckets
+aws s3 rm s3://emergentworks.org/ --recursive
```
Comment (markthom-as) [inline deploy-website.sh:7]: We might want to check for the existence of "aws" here and send an explicit failure/exit signal here... Not sure how github actions will react in that case.

File: package.json [hunk 5]
```diff
@@ -6,6 +6,8 @@
   "scripts": {
     "build": "gatsby build",
     "develop": "gatsby develop",
+    "deploy": "bash deploy-website.sh",
+    "update": "aws cloudfront create-invalidation --distribution-id E3NHW48S3W8DOR --paths '/*'",
```
Comment (markthom-as) [inline package.json:10]: I think moving this to a script with checks for the "aws" binary as mentioned above would be great. I think you could even have both of these scripts in the same shell file and have this one activated by a flag.

I think it's best practice as well to keep any of the specific values (s3 bucket URL, cloudfront distro ID) as variables (eventually env variables) at the top, to make the script more portable and easily updatable.

**Review by markthom-as — CHANGES_REQUESTED** [review]
This is awesome!!!!!
Left a few questions and suggestions for changes, but nothing that'd be blocking for deploy if you don't have the time to implement them.

Things you don't have the time to update I just ask that you create tickets for in the backlog so we can get them done in the future.

Great work!!!

File: deploy-website.sh (hunk 3 above)
Comment (elleneub) [reply deploy-website.sh]: switch to exit on failure, good call

File: package.json (hunk 5 above)
Comment (elleneub) [reply package.json]: moved and put environment vars in a .env file

**Review by markthom-as — APPROVED** [review]
LGTM, great work!

**Pull request merged**